modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java

```diff
@@ -1063,18 +1064,30 @@ public void removeIndex(Session session, Index h2Idx) {
                 if (idx == targetIdx || (idx instanceof GridH2ProxyIndex &&
                     ((GridH2ProxyIndex)idx).underlyingIndex() == targetIdx)) {
 
-                    idxs.remove(i);
+                    Index idx0 = idxs.remove(i);
 
-                    if (idx instanceof GridH2ProxyIndex &&
+                    if (idx0 instanceof GridH2ProxyIndex &&
                         idx.getSchema().findIndex(session, idx.getName()) != null)
                         database.removeSchemaObject(session, idx);
 
+                    GridCacheContext cctx0 = cacheInfo.cacheContext();
+
+                    if (cctx0 != null && idx0 instanceof GridH2IndexBase) {
+                        cctx0.shared().database().checkpointReadLock();
+
+                        try {
+                            ((GridH2IndexBase)idx0).destroy(rmIndex);
```

Comment:
Why do we need to pass `rmIndex` parameter here? In what cases we expect `rmIndex == false` here? Can we pass `true` unconditionally?

Reply:
cause on node and further cache.stop we need to set _false_ here, that by design and beyond  the scope of this ticket. If we pass almost true we would have undeterminate pageStore exceptions here )

Reply:
Thank, got it.